Финальное задание 4ого спринта.

SanatinaKate opened the pull request:
Sprint_4

SanatinaKate committed:
Финальное задание 4ого спринта.

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,228 @@
+
+        # получаем жанр каждой добавленной книги и проверяем правильность реультата
+        for name, genre in books.items():
+            genre_expected = genre if genre in collector.genre else ''
```
Нужно исправить: условий в тестах быть не может, тест всегда однозначен.

SanatinaKate replied:
@AlexKitIt В тестах я использую if для формирования ожидаемого результата. Можно ли так делать и нужно ли это исправлять?

AlexKitIt replied:
По логике получается, что мы не знаем как именно пройдет и закончится тест. Такого быть не должно. Нам либо нужны отдельные объекты книг где есть только верные книги и жанры, либо добавлять в тестах только нужные данные

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,228 @@
+            books_expected[name] = ''
+
+        # получаем словарь с пустыми жанрами добавленных книг и проверяем правильность результата
+        assert collector.get_books_genre() == books_expected
```
Почему бы тет просто не установить атрибут и не проверить что метод вернет именно этот атрибут

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,24 @@
+
+
+# Создание тестового словаря с книгами
+@pytest.fixture(scope="class")
+def books(scope="class"):
+    books = {'Горе от ума': 'Комедии',
+             'Десять негритят': 'Детективы',
+             'Дракула': 'Ужасы',
+             'Дюна': 'Фантастка',
+             'Ревизор': 'Комедии',
+             'Оно': 'Ужасы',
+             'Cнеговик': 'Детективы',
+             'Трое из Простоквашино': 'Мультфильмы',
+             'Тысяча и одна ночь': 'Сказки',
+             'Чайка': 'Комедии'}
+    return books
```
Нужно исправить: фикстуры не занимаются прокидыванием данных в тест и выполнением примитивной логики, они выполняют сложную логику предусловий\постусловий и вычислений. Статические данные необходимо хранить в data модуле и оттуда просто импортировать куда необходимо

SanatinaKate replied:
@AlexKitIt Будет ли достаточно создать отдельный файл с классом, в котором хранить тестовые данные с книгами и потом импортировать этот класс в файл с тестами?

AlexKitIt replied:
Именно так и необходимо сделать. Модуль назвать data и там разместить всю статику

SanatinaKate committed:
Финальное задание 4ого спринта - исправление замечаний:
1. В тестах выполнен рефакторинг, убранны все операторы if.
2. Статические данные перенесены из фикстуры в отдельный файл (класс).
